src/Http/Headers/src/CookieHeaderValue.cs

```diff
@@ -164,10 +164,8 @@ public static bool TryParseStrictList(IList<string>? inputs, [NotNullWhen(true)]
 
     internal static void CheckNameFormat(StringSegment name, string parameterName)
     {
-        if (name == null)
-        {
-            throw new ArgumentNullException(nameof(name));
-        }
+        ArgumentNullException.ThrowIfNull(name, parameterName);
```

Comment:
This is covered by the line below, and the implicit equals operator on the StringSegment struct, I think? Anyway in other cases you don't have this line.
```suggestion
```